### PR TITLE
REF: replace column-wise, remove BlockManager.apply filter kwarg

### PR DESCRIPTION
**Pull request by jbrockmendel**
Follow-up to #33279.

### CODE REVIEW

File: pandas/core/generic.py [hunk 1]
```diff
@@ -6550,18 +6549,20 @@ def replace(
...
-                            convert=convert,
-                        )
+
+                    # Operate column-wise
+                    res = self if inplace else self.copy()
+                    ax = self._info_axis
+                    # Note: we only get here with self.ndim == 2
+                    for i in range(len(ax)):
+                        if ax[i] in to_replace:
+                            ser = self._ixs(i, axis=1)
```
Comment (jreback) [inline pandas/core/generic.py:6559]: i would just use `.iloc` here

File: pandas/core/generic.py [hunk 2]
```diff
@@ -6602,17 +6603,20 @@ def replace(
...
-                                regex=regex,
-                            )
+
+                    # Operate column-wise
+                    res = self if inplace else self.copy()
+                    ax = self._info_axis
+                    for i in range(len(ax)):
+                        if ax[i] in value:
+                            v = value[ax[i]]
+                            ser = self._ixs(i, axis=1)
```
Comment (jreback) [inline pandas/core/generic.py:6613]: same,

can you factor this to a function instead.

**Review by jreback — CHANGES_REQUESTED** [review]

File: pandas/core/generic.py (hunk 2 above)
Comment (jbrockmendel) [reply pandas/core/generic.py]: refactored + green

i think in follow-up might be able to use this new helper method in one or two more places

File: pandas/tests/series/methods/test_replace.py [hunk 3]
```diff
@@ -373,3 +373,19 @@ def test_replace_invalid_to_replace(self):
         )
         with pytest.raises(TypeError, match=msg):
             series.replace(lambda x: x.strip())
+
+    def test_replace_only_one_dictlike_arg(self):
+        # GH#33340
```
Comment (jreback) [inline pandas/tests/series/methods/test_replace.py:378]: just increasing test coverage? or is this an actual bug?

**Review by jreback — APPROVED** [review]
lgtm. a question.

File: pandas/tests/series/methods/test_replace.py (hunk 3 above)
Comment (jbrockmendel) [reply pandas/tests/series/methods/test_replace.py:378]: the new helper method is DataFrame-only, so this PR adds a if-series-then-raise check.  this test hits that

File: pandas/core/frame.py [hunk 4]
```diff
@@ -4025,6 +4025,41 @@ def replace(
             method=method,
         )
 
+    def _replace_columnwise(
```
Comment (jreback) [inline pandas/core/frame.py:4028]: i think may make sense to move this somewhere, maybe missing.py just to try to reduce size in this module. anyhow for another day.

**Comment by jreback** [comment]
thanks

**Pull request merged**